fecgov/fec-cms
# Specify public (client-side) and private (server-side) keys

## Summary (required)

- Resolves https://github.com/fecgov/fec-accounts/issues/176: Specify public (client-side) and private (server-side) keys

- Add **new** env vars for `FEC_WEB_API_KEY_PRIVATE` (Don't replace `FEC_WEB_API_KEY` until the code is live)
- Put in follow-up issue to remove `FEC_WEB_API_KEY` env vars after deployments and research possible limits per IP for `FEC_WEB_API_KEY_PUBLIC`, add them via API umbrella, make sure `stage` and `prod` are using different keys https://github.com/fecgov/fec-cms/issues/3029

## Impacted areas of the application
List general components of the application that this PR will affect:

-  Public (client-side) and private (server-side)  API calls

## How to test

- Run this branch locally

**Option 1: Test with real keys (tests whether the keys are being passed and accepted)**
- set `FEC_WEB_API_KEY_PRIVATE`. You can probably use what you had previously saved in `FEC_WEB_API_KEY` (`export FEC_WEB_API_KEY_PRIVATE=$FEC_WEB_API_KEY`)
- Connect to `stage` API (https://api-stage.open.fec.gov)
- Check terminal output for python/server-side calls to make sure they're using the key saved in `FEC_WEB_API_KEY_PRIVATE`
- Check network tab of browser for JS/client-side calls to make sure they're using the key saved in `FEC_WEB_API_KEY_PUBLIC`

**Option 2: Test with placeholder keys (easier to see mistakes, won't actually test the keys)**
- `export FEC_WEB_API_KEY_PRIVATE="PRIVATE_KEY"` and `export FEC_WEB_API_KEY_PUBLIC="PUBLIC_KEY"` 
- Connect to `dev` API (https://fec-dev-api.app.cloud.gov) - doesn't have API umbrella
- Check terminal output for python/server-side calls to make sure they're using the key saved in `PRIVATE_KEY`
- Check network tab of browser for JS/client-side calls to make sure they're using the key `PUBLIC_KEY`

**Example server-side calls**
- http://localhost:8000/data/candidate/P60007168/?cycle=2020&election_full=true
- http://localhost:8000/data/legal/search/advisory-opinions/?type=advisory_opinions&ao_no=2019-01&ao_category=F

**Example client-side calls**
- http://localhost:8000/data/receipts/
- http://localhost:8000/data/elections/president/2020/
- http://localhost:8000/legal-resources/enforcement/audit-search/?primary_category_id=5
- http://localhost:8000/data/candidate/P60007168/?cycle=2020&election_full=true&tab=raising
